File: leetcode/must-do-easy-questions/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int major = -1;
        int count = 0;
        for(int i = 0; i < nums.size(); i++) {
            if (i == 0) {
                major = nums[i];
                count = 1;
                continue;
            }
            (nums[i] != major) ? count-- : count++;
            if (count < 0) {
                major = nums[i];
                count = 1;
            }
        }
        return major;
    }
};

File: leetcode/30-day-leetcoding-challenge-May/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const majorityElement = function(nums) {
    let major;
    let cnt = 0;
    for(const n of nums) {
        if (cnt === 0) {
            major = n;
        }
        cnt += major === n ? 1 : -1;
    }
    return major;
};

// [1,1,2,2,2,3,3,3,3,3,3,3,3,3] -> 3
